### PR TITLE
make tensora compilable on Darwin

### PR DESCRIPTION
**Pull request by amirmolavi**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import subprocess
 from pathlib import Path
+import platform
```
Comment (drhagenabm) [inline setup.py:3]:
```suggestion
import platform
import subprocess
from pathlib import Path
```

File: setup.py [hunk 2]
```diff
@@ -14,11 +15,19 @@
...
+            install_path = r'-DCMAKE_INSTALL_RPATH=@loader_path/../lib'
+        else:
+            raise NotImplementedError
+
         taco_build_dir.mkdir(parents=True, exist_ok=True)
         subprocess.check_call(['cmake', str(taco_source_dir),
                                '-DCMAKE_BUILD_TYPE=Release',
                                f'-DCMAKE_INSTALL_PREFIX={taco_install_dir}',
-                               r'-DCMAKE_INSTALL_RPATH=\$ORIGIN/../lib'],  # TODO: Will only work on Linux
+                               install_path],  # TODO: Will only work on Linux
```
Comment (drhagenabm) [inline setup.py:30]:
```suggestion
                               install_path],
```

File: setup.py [hunk 3]
```diff
@@ -14,11 +15,19 @@
 class TensoraBuild(build):
     def run(self):
         # Build taco
+        OS = platform.system()
+        if OS == "Linux":
+            install_path = r'-DCMAKE_INSTALL_RPATH=\$ORIGIN/../lib'
+        elif OS == "Darwin":
+            install_path = r'-DCMAKE_INSTALL_RPATH=@loader_path/../lib'
+        else:
+            raise NotImplementedError
```
Comment (drhagenabm) [inline setup.py:24]:
```suggestion
            raise NotImplementedError(f'Tensora cannot be installed on {OS}')
```

**Review by drhagenabm — CHANGES_REQUESTED** [review]
Thanks for getting this working on Mac. I just have a few stylistic suggestions.

File: setup.py (hunk 1 above)
Comment (drhagen) [inline setup.py:3]:
```suggestion
import platform
import subprocess
from pathlib import Path
```

File: setup.py (hunk 3 above)
Comment (drhagen) [inline setup.py:24]:
```suggestion
            raise NotImplementedError(f'Tensora cannot be installed on {OS}')
```

File: setup.py (hunk 2 above)
Comment (drhagen) [inline setup.py:30]:
```suggestion
                               install_path],
```

**Review by drhagen — CHANGES_REQUESTED** [review]
Thanks for getting this working on Mac. I just have a few stylistic suggestions.

**Pull request merged**